installers/MacInstaller.hs

```diff
@@ -16,101 +16,76 @@ module MacInstaller
+import           Control.Monad (unless)
+import           Data.Text (Text)
 import qualified Data.Text as T
 import           System.Directory (copyFile, createDirectoryIfMissing, doesFileExist, renameFile)
-import           System.Environment (lookupEnv)
 import           System.FilePath ((</>), FilePath)
 import           System.FilePath.Glob (glob)
 import           Filesystem.Path.CurrentOS (encodeString)
-import           Turtle (ExitCode (..), echo, proc, procs, which, Managed, with)
+import           Text.Printf (printf)
+import           Turtle (ExitCode (..), echo, proc, procs, shells, which, Managed, with)
 import           Turtle.Line (unsafeTextToLine)
 
 import           RewriteLibs (chain)
 
 import           System.IO (hSetBuffering, BufferMode(NoBuffering))
 
-data InstallerConfig = InstallerConfig {
-    icApi :: String
-  , appNameLowercase :: T.Text
-  , appName :: String
-  , pkg :: T.Text
-  , predownloadChain :: Bool
-  , appRoot :: String
-}
-
--- In both Travis and Buildkite, the environment variable is set to
--- the pull request number if the current job is a pull request build,
--- or "false" if it’s not.
-pullRequestFromEnv :: IO (Maybe String)
-pullRequestFromEnv = liftM2 (<|>) (getPR "BUILDKITE_PULL_REQUEST") (getPR "TRAVIS_PULL_REQUEST")
-  where
-    getPR = fmap interpret . lookupEnv
-    interpret Nothing        = Nothing
-    interpret (Just "false") = Nothing
-    interpret (Just num)     = Just num
+import           Config
+import           Types
 
-travisJobIdFromEnv :: IO (Maybe String)
-travisJobIdFromEnv = lookupEnv "TRAVIS_JOB_ID"
+
 
-installerConfigFromEnv :: IO InstallerConfig
-installerConfigFromEnv = mkEnv <$> envAPI <*> envVersion
-  where
-    envAPI = fromMaybe "cardano" <$> lookupEnv "API"
-    envVersion = fromMaybe "dev" <$> lookupEnv "DAEDALUS_VERSION"
-    mkEnv "cardano" ver = InstallerConfig
-      { icApi = "cardano"
-      , predownloadChain = False
-      , appNameLowercase = "daedalus"
-      , appName = "Daedalus"
-      , pkg = "dist/Daedalus-installer-" <> T.pack ver <> ".pkg"
-      , appRoot = "../release/darwin-x64/Daedalus-darwin-x64/Daedalus.app"
-      }
-
-main :: IO ()
-main = do
+main :: Options -> IO ()
+main opts@Options{..} = do
   hSetBuffering stdout NoBuffering
 
-  cfg <- installerConfigFromEnv
-  tempInstaller <- makeInstaller cfg
-  shouldSign <- shouldSignDecision
+  let appRoot = "../release/darwin-x64/Daedalus-darwin-x64/Daedalus.app"
+
+  echo "Generating configuration file:  launcher-config.yaml"
+  generateConfig (Request Macos64 oCluster Launcher) "./dhall" "launcher-config.yaml"
+  echo "Generating configuration file:  wallet-topology.yaml"
+  generateConfig (Request Macos64 oCluster Topology) "./dhall" "wallet-topology.yaml"
+
+  echo "Packaging frontend"
+  shells "npm run package -- --icon installers/icons/256x256" mempty
+
+  tempInstaller <- makeInstaller opts appRoot
 
-  if shouldSign
+  let shouldSign = oPullReq == Nothing
```

Comment:
This changed behaviour - previously it would sign on buildkite regardless if it was a PR or not.

Reply:
Do you think we should revert that to always sign on Buildkite?

Reply:
This PR shouldn't change/touch this behavior since it's out of scope.

If it did, we should discuss it - and for now we sign all buildkite builds since noone except trusted IOHK members can trigger them.

Reply:
Done.